src/Avalonia.Native/Metal.cs

```diff
@@ -84,8 +85,15 @@ public void Dispose()
 
     public IMetalPlatformSurfaceRenderingSession BeginRendering()
     {
-        var session = _native.BeginDrawing();
-        return new MetalDrawingSession(session);
+        try
+        {
+            var session = _native.BeginDrawing();
+            return new MetalDrawingSession(session);
+        }
+        catch (COMException com) when (com.HResult == (int)AvnResultCodes.E_AVN_RENDER_TARGET_NOT_READY)
```

Comment:
Minor. But would it be possible to return HResult from BeginDrawing, instead of throwing COMException?
Just to avoid unfilterable exceptions noise.
Throwing RenderTargetNotReadyException exception is fine, as it can be easily filtered-out by the debugger by type.

Reply:
We can't have non-HRESULT returns for methods with `out` object arguments. Not supported by MicroCOM and would require some extensive refactoring of the way argument marshaling works.